collector/mdadm_linux.go

```diff
@@ -225,7 +231,7 @@ func parseMdstat(mdStatusFilePath string) ([]mdStatus, error) {
 			syncedBlocks = size
 		}
 
-		mdStates = append(mdStates, mdStatus{currentMD, isActive, active, total, size, syncedBlocks})
+		mdStates = append(mdStates, mdStatus{currentMD, isActive, active, failed, spare, total, size, syncedBlocks})
```

Comment:
The `mdStatus` struct is becoming a bit too long to keep all fields in mind and not assigning them in the wrong order by accident. I highly vote to use named fields here:

```go
mdStatus{
    name: currentMD,
    active: isActive,
    disksFailed: failed,
    ...
}
```